test/Mono.Linker.Tests/TestCasesRunner/TestCaseCompiler.cs

```diff
@@ -185,6 +191,101 @@ protected NPath CompileAssembly (CompilerOptions options)
 
 		protected virtual NPath CompileCSharpAssemblyWithDefaultCompiler (CompilerOptions options)
 		{
+#if NETCOREAPP
```

Comment:
I added the in-process compiler mostly for two reasons:
- to match the default behavior of codedom
- because running csc in its own process isn't a common .NET Core scenario. In fact I wasn't sure if we even supported it, until I poked around and saw that we have a csc.dll in our SDK directory that can be run via "dotnet /path/to/csc.dll". The good news is that this dll has a runtimeconfig that specifies server GC!

I'll try to get these tests working with the out-of-process (+server GC) approach and see what the perf is like. To make it a fair comparison, I'll try using server GC for the in-process version as well.
Thanks!

Reply:
> to match the default behavior of codedom

I only used that for convenience and to avoid having to find `csc` on Windows.  Later on it turned out we had to find `csc` anyways.

> I'll try to get these tests working with the out-of-process (+server GC) approach and see what the perf is like. To make it a fair comparison, I'll try using server GC for the in-process version as well.
👍

Reply:
If it's fine with you, I'd like to do this investigation in a follow-up to this PR. At the moment my priority is to get the tests working on .NET Core and running in ci.

Reply:
Filed https://github.com/mono/linker/issues/499 to track.